In [1]:
import csv
f = open("guns.csv", "r")
csvreader = csv.reader(f)
data = list(csvreader)
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
headers = data[0:1]
data = data[1:]
print(headers)
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']]
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [3]:
years = [row[1] for row in data]
years_counts = {}
for elt in years:
    if elt in years_counts:
        years_counts[elt] += 1
    else:
        years_counts[elt] = 1

print(years_counts)

{'2012': 33563, '2014': 33599, '2013': 33636}


In [4]:
import datetime

dates = [datetime.datetime(int(row[1]), int(row[2]), day=1) for row in data]


print(dates[:5])  

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [5]:
date_counts = {}
for elt in dates:
    if elt not in date_counts:
        date_counts[elt] = 1
    else:
        date_counts[elt] += 1
date_counts
     

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

In [6]:
sex = [row[5] for row in data]
sex_counts = {}
for gender in sex:
    if gender in sex_counts:
        sex_counts[gender] += 1
    else:
        sex_counts[gender] = 1
sex_counts

{'F': 14449, 'M': 86349}

In [7]:
races = [row[7] for row in data]
race_counts = {}
for race in races:
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

Gun Deaths in the US:
The highest death rate is in July, 2013 (3079), whereas, the lowest rate is 2357 in Feb, 2012. Comparing the gender, fatality of male is 6 times higher than female. However, the white has the highest fatality and the native american has the lowest, which is 66237 and 917, respectively.

In [8]:
import csv
f = open("census.csv", "r")
csvreader = csv.reader(f)
census = list(csvreader)
census

[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

In [12]:
mapping = {
 'Asian/Pacific Islander': 15159516 + 674625,
 'Black': 40250635,
 'Native American/Native Alaskan': 3739506,
 'Hispanic': 44618105,
 'White': 197318956,
}
race_per_hundredk = {}
for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000
    
race_per_hundredk

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

In [18]:
intents = [row[3] for row in data]
races = [row[7] for row in data]
homicide_race_counts = {}
for i, race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 1
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1
    
homicide_per_hundredk = {}
for k,v in homicide_race_counts.items():
    homicide_per_hundredk[k] = (v / mapping[k]) * 100000

homicide_race_counts
homicide_per_hundredk


{'Asian/Pacific Islander': 3.5366616982885275,
 'Black': 48.47376942003524,
 'Hispanic': 12.62940234687242,
 'Native American/Native Alaskan': 8.744470526320857,
 'White': 4.636148591826119}

In [19]:
homicide_month_counts = {}
for i, month in enumerate(dates):
    if month not in homicide_month_counts:
        homicide_month_counts[month] = 1
    if intents[i] == "Homicide":
        homicide_month_counts[month] += 1
homicide_month_counts

{datetime.datetime(2012, 1, 1, 0, 0): 973,
 datetime.datetime(2012, 2, 1, 0, 0): 750,
 datetime.datetime(2012, 3, 1, 0, 0): 967,
 datetime.datetime(2012, 4, 1, 0, 0): 1000,
 datetime.datetime(2012, 5, 1, 0, 0): 1004,
 datetime.datetime(2012, 6, 1, 0, 0): 1045,
 datetime.datetime(2012, 7, 1, 0, 0): 1161,
 datetime.datetime(2012, 8, 1, 0, 0): 1091,
 datetime.datetime(2012, 9, 1, 0, 0): 1071,
 datetime.datetime(2012, 10, 1, 0, 0): 980,
 datetime.datetime(2012, 11, 1, 0, 0): 979,
 datetime.datetime(2012, 12, 1, 0, 0): 1084,
 datetime.datetime(2013, 1, 1, 0, 0): 987,
 datetime.datetime(2013, 2, 1, 0, 0): 722,
 datetime.datetime(2013, 3, 1, 0, 0): 924,
 datetime.datetime(2013, 4, 1, 0, 0): 917,
 datetime.datetime(2013, 5, 1, 0, 0): 956,
 datetime.datetime(2013, 6, 1, 0, 0): 1067,
 datetime.datetime(2013, 7, 1, 0, 0): 1138,
 datetime.datetime(2013, 8, 1, 0, 0): 1001,
 datetime.datetime(2013, 9, 1, 0, 0): 955,
 datetime.datetime(2013, 10, 1, 0, 0): 1010,
 datetime.datetime(2013, 11, 1, 0, 0): 